test/parser_test.cpp

```diff
@@ -34,17 +34,8 @@ TEST(parserTest, test_iString)
 
 TEST(parserTest, test_invalidInput)
 {
-    const std::string expectedError = "Invalid key: nev";
     std::ifstream invalidInput("units/invalid_unit.json");
-    try
-    {
-        std::map<std::string, std::string> data = JSONParser::Parse(invalidInput);
-        FAIL() << "Expected std::runtime_error";
-    }
-    catch (std::runtime_error &e)
-    {
-        ASSERT_EQ(e.what(), expectedError);
-    }
+    ASSERT_THROW(JSONParser::ParseStream(invalidInput), std::runtime_error);
```

Comment:
Kicsit már sajnálom is, hogy ezt átírattam, mert így meg a hibaüzenetet nem lehet pontosan tesztelni.